pkg/diagnose/cni.go

```diff
@@ -206,3 +222,101 @@ func mustHaveSubmariner(clusterInfo *cluster.Info) {
 		panic("cluster.Info.Submariner field cannot be nil")
 	}
 }
+
+func checkOVNVersion(info *cluster.Info, status reporter.Interface) bool {
+	status.Start("Checking OVN version")
+	defer status.End()
+
+	clientSet := info.ClientProducer.ForKubernetes()
+
+	ovnPod, err := mustFindPod(clientSet, ovnKubeDBPodLabel)
+	if err != nil {
+		status.Failure("Failed to get OVNKubeDB Pod %v", err)
+		return false
+	}
+
+	ovnNBVersion, err := getOVNNBVersion(clientSet, info.RestConfig, ovnPod)
+	if err != nil {
+		status.Failure("Failed to get OVN NB version %v", err)
+		return false
+	}
+
+	if version.Compare(ovnNBVersion, minOVNNBVersion, "<") {
+		status.Failure("The OVN NB DB version %v is less than the minimum supported version %v", ovnNBVersion, minOVNNBVersion)
+		return false
+	}
+
+	status.Success("The OVN NB DB version %v is supported", ovnNBVersion)
+
+	return true
+}
+
+func mustFindPod(clientSet kubernetes.Interface, labelSelector string) (*corev1.Pod, error) {
+	pods, err := clientSet.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{
+		LabelSelector: labelSelector,
+		Limit:         1,
+	})
+	if err != nil {
+		return nil, errors.WithMessagef(err, "error listing Pods by label selector %q", labelSelector)
+	}
+
+	if len(pods.Items) == 0 {
+		return nil, errors.WithMessagef(err, "no pod found with label %v", labelSelector)
+	}
+
+	return &pods.Items[0], nil
+}
+
+func getOVNNBVersion(clientSet kubernetes.Interface, config *rest.Config, pod *corev1.Pod) (string, error) {
+	containerName := ""
+
+	for i := 0; i < len(pod.Spec.Containers); i++ {
+		container := pod.Spec.Containers[i]
+		// NBDB container name is nb-ovsdb [vanilla OVNK] or nbdb [OCP].
+		if strings.HasPrefix(container.Name, "nb") {
+			containerName = container.Name
+		}
+	}
+
+	cmd := []string{"ovn-nbctl", "-V"}
+	req := clientSet.CoreV1().RESTClient().Post().
+		Resource("pods").
+		Name(pod.Name).
+		Namespace(pod.Namespace).
+		SubResource("exec").
+		Param("container", containerName)
+	req.VersionedParams(&corev1.PodExecOptions{
+		Container: containerName,
+		Command:   cmd,
+		Stdin:     false,
+		Stdout:    true,
+		Stderr:    true,
+		TTY:       false,
+	}, scheme.ParameterCodec)
+
+	var stdout, stderr bytes.Buffer
+
+	exec, err := remotecommand.NewSPDYExecutor(config, "POST", req.URL())
+	if err != nil {
+		return "", errors.WithMessagef(err, "failed to create SPDY executor")
+	}
+
+	err = exec.Stream(remotecommand.StreamOptions{
+		Stdin:  nil,
+		Stdout: &stdout,
+		Stderr: &stderr,
+		Tty:    false,
+	})
+
+	if err != nil {
+		return "", errors.WithMessagef(err, "failed to execute SPDY command")
+	}
+
+	results := strings.Split(stdout.String(), "DB Schema")
+
+	if len(results) < 2 {
+		return "", errors.WithMessagef(err, "OVN NB DB Schema version not found")
```

Comment:
Minor thing but if this occurred it would be useful to see the output.

```suggestion
		return "", errors.WithMessagef(err, "unable to determine the version from the ovn-nbctl output: %q", stdout.String())
```